tests/test_commons_auth.py

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+import logging
+import sys
+import unittest
+from contextlib import contextmanager
+
+import connexion.apis.abstract
+import os
+import requests
+
+pkg_root = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))  # noqa
+sys.path.insert(0, pkg_root)  # noqa
+
+import dss
+from dss.logging import configure_test_logging
+from dss.util import UrlBuilder
+from tests import get_auth_header
+from tests.infra.server import ThreadedLocalServer
+from dss.config import override_bucket_config, BucketConfig, Replica
+from tests.infra import DSSAssertMixin, DSSUploadMixin, get_env, testmode
+
+logger = logging.getLogger(__name__)
+
+
+def setUpModule():
+    configure_test_logging()
+
+
+@testmode.standalone
+class TestCommonsAuth(unittest.TestCase, DSSAssertMixin, DSSUploadMixin):
+    @classmethod
+    def setUpClass(cls):
+        cls.app = ThreadedLocalServer()
+        cls.app.start()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.app.shutdown()
+
+    def construct_url(self, path: str, replica: Replica) -> str:
+        version = '2017-06-20T214506.766634Z'
+        return str(UrlBuilder()
+                   .set(path=path)
+                   .add_query('replica', replica.name)
+                   .add_query('version', version))
+
+    def launch_checkout(self, url: str, replica: Replica) -> str:
```

Comment:
I think the testing of checkout can/should be simplified, and the testing improved by doing so.
I think right now we are not actually testing authentication for `/bundles/{uuid}/checkout`,
and I think a dummy/mock "arn" aka `checkout_job_id` (which is just a UUID) can be passed to `/bundles/checkout/{checkout_job_id}` to test that.

Reply:
Added.